src/main/java/org/assertj/core/api/AbstractThrowableAssert.java

```diff
@@ -203,6 +203,31 @@ public SELF hasNoCause() {
     return new ThrowableAssert<>(org.assertj.core.util.Throwables.getRootCause(actual));
   }
 
+  /**
+   * Returns a new list assertion of every cause in the chain of causes.
+   * The list is ordered from the current {@link Throwable} up to the root cause.
+   * <p>
+   * Examples:
+   * <pre><code class='java'> Throwable rootCause = new JdbcException("invalid query");
+   * Throwable cause =  new RuntimeException("some failure", rootCause);
+   * Throwable throwable = new Exception("boom!", cause);
+   *
+   * // typical use:
+   * assertThat(throwable).asCausesChain()
+   *                      .extracting(Throwable::getMessage)
+   *                      .anyMatch("some failure");</code></pre>
+   *
+   * @return a new assertion object
+   * @throws AssertionError if the actual {@code Throwable} is {@code null}.
+   * @throws AssertionError if the actual {@code Throwable} does not have a cause.
```

Comment:
I'm not sure if we should relax this and return an empty list instead.

Reply:
It's not an obvious call  but I think I'm in favor of returning an empty list.

pros: 
- you can write `assertThat(throwable).causesChain().isEmpty();` which is reasonable
- in API I tend to prefer returning empty collection over null/failure

cons: 
- if we return an empty list then some assertions like `allMatch` will pass (it succeeds for any empty list)
- users probably expect to have causes some failing if there are none might be better

The last point is not a real issue IMO as the assertions applied on the list of causes will most probably fail if we return an empty list (except for a few like `allMatch`).

TLDR: let's go with returning an empty list ;-)

Reply:
Modified with the empty list